docker/nginx/libs/utils.spec.lua

```diff
@@ -77,3 +79,125 @@ describe("extract_cookie_value", function()
+        os.getenv.on_call_with("foo").returns("bar")
+
+        assert.is_equal(utils.getenv("foo"), "bar")
+    end)
+
+    describe("parse", function()
+        it("should throw on not supported parser", function()
+            os.getenv.on_call_with("foo").returns("test")
+
+            assert.has_error(function()
+                utils.getenv("foo", "starwars")
+            end, "utils.getenv: Parsing to 'starwars' is not supported")
+        end)
+
+        describe("as boolean", function()
+            it("should return nil for not existing env var", function()
+                os.getenv.on_call_with("foo").returns(nil)
+
+                assert.is_nil(utils.getenv("foo", "boolean"))
+            end)
+
+            it("should return nil for env var that is an empty string", function()
+                os.getenv.on_call_with("foo").returns("")
+
+                assert.is_nil(utils.getenv("foo", "boolean"))
+            end)
+
+            it("should parse 'true' string as true", function()
+                os.getenv.on_call_with("foo").returns("true")
+
+                assert.is_true(utils.getenv("foo", "boolean"))
+            end)
+
+            it("should parse '1' string as true", function()
+                os.getenv.on_call_with("foo").returns("1")
+
+                assert.is_true(utils.getenv("foo", "boolean"))
+            end)
+
+            it("should parse 'false' string as false", function()
+                os.getenv.on_call_with("foo").returns("false")
+
+                assert.is_false(utils.getenv("foo", "boolean"))
+            end)
+
+            it("should parse '0' string as false", function()
+                os.getenv.on_call_with("foo").returns("0")
+
+                assert.is_false(utils.getenv("foo", "boolean"))
+            end)
+
+            it("should throw an error for not supported string", function()
+                os.getenv.on_call_with("foo").returns("mandalorian")
+
+                assert.has_error(function()
+                    utils.getenv("foo", "boolean")
+                end, "utils.getenv: Parsing value 'mandalorian' to boolean is not supported")
+            end)
+        end)
+
+        describe("as integer", function()
+            it("should return nil for not existing env var", function()
+                os.getenv.on_call_with("foo").returns(nil)
+
+                assert.is_nil(utils.getenv("foo", "integer"))
+            end)
+
+            it("should return nil for env var that is an empty string", function()
+                os.getenv.on_call_with("foo").returns("")
+
+                assert.is_nil(utils.getenv("foo", "integer"))
+            end)
+
+            it("should parse '0' string as 0", function()
+                os.getenv.on_call_with("foo").returns("0")
+
+                assert.equals(utils.getenv("foo", "integer"), 0)
+            end)
+
+            it("should parse '1' string as 1", function()
```

Comment:
not sure whether this is valuable to have as a unit test, seems like these are a bit redundant

Reply:
what I wanted to achieve is to test positive/zero/negative examples

Reply:
if only there was a way to write these types of explanations in the code...

Something like a comment 😜

Reply:
> if only there was a way to write these types of explanations in the code...
> 
> Something like a comment 😜

I'll add a comment on some tests that need explanation in a follow up when writing tests for account.lua